sop-extras/src/Data/SOP/InPairs.hs

```diff
@@ -99,6 +105,15 @@ hcpure _ f =
   RequiringBoth
 -------------------------------------------------------------------------------}
 
+infixr 2 :..:
```

Comment:
I think these are more flexible and easier to read/track than `Requiring` etc.

TODO we should replace all remaining usings of `Requiring` etc